File: Day 1/C/PartOne.c
#include <stdio.h>
#include <ctype.h>

int main()
{
    char *filepath = "..\\input.txt";
    FILE *input_file = fopen(filepath, "r");
    char input_line[128];
    int sum = 0;
    while (fgets(input_line, 128, input_file) != NULL) {
        int calibration_value = 0;
        int last_digit = 0;
        for (int i=0; input_line[i] != '\0'; i++) {
            if (isdigit(input_line[i])) {
                last_digit = input_line[i] - '0';
                if (calibration_value == 0) {
                    calibration_value = (last_digit * 10);
                }
            }
        }
        sum = sum + calibration_value + last_digit;
    }

    printf("\n%d", sum);

    fclose(input_file);
    return 0;
}

File: Day 1/Python/PartOne.py
"""
Getting the first and last digit in a string
"""

def get_calibration_value(word:str):
    calibration_value = 0
    last_digit = ""
    for value in word:
        if value.isdigit():
            value = int(value)
            if last_digit == "":
                calibration_value += (value * 10)
            last_digit = value
    calibration_value += last_digit
    return calibration_value


def main(input_filepath):
    input_file = open(input_filepath)
    file_text = input_file.readlines()
    sum_of_calibration_values = 0
    for word in file_text:
        calibration_value = get_calibration_value(word)
        sum_of_calibration_values += calibration_value
    print(sum_of_calibration_values)


main("Day 1\input.txt")